### ISSUE TITLE
appservice kube - No module named 'azure.mgmt.web.v2021_01_01'

### ISSUE THREAD

**Issue by likamrat** [Azure/azure-cli#24744]
**Related command**
az appservice kube

**Describe the bug**
When running the _az appservice kube_ command, getting the following output.

```powershell
ERROR: The command failed with an unexpected error. Here is the traceback:
ERROR: No module named 'azure.mgmt.web.v2021_01_01'
Traceback (most recent call last):
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\knack/cli.py", line 233, in invoke
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 663, in execute
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 726, in _run_jobs_serially
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 697, in _run_job
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 333, in __call__
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/command_operation.py", line 363, in handler
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/arm.py", line 429, in show_exception_handler
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/command_operation.py", line 361, in handler
  File "C:\Users\arcdemo\.azure\cliextensions\appservice-kube\azext_appservice_kube\custom.py", line 305, in show_kube_environments
    client = _get_kube_client(cmd)
  File "C:\Users\arcdemo\.azure\cliextensions\appservice-kube\azext_appservice_kube\custom.py", line 301, in _get_kube_client
    return client.kube_environments
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/mgmt/web/_web_site_management_client.py", line 627, in kube_environments
ModuleNotFoundError: No module named 'azure.mgmt.web.v2021_01_01'
```
<img width="1090" alt="Snag_28fc08b1" src="https://user-images.githubusercontent.com/46730876/204014966-0c207332-d590-42c1-95ae-62363ae9a4f8.png">

**To Reproduce**

### tl;dr
Run the the _az appservice kube_ command in Windows PoSH (not tested in Bash but it looks like it will happen there as well)

### Long version
Follow the [Deploy an App Service app using custom container on AKS using an ARM Template](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_app_svc/aks/aks_app_service_arm_template) Azure Arc Jumpstart scenario

**Expected behavior**
_az appservice kube_ command should result in no Python dependency errors

**Environment summary**
Az CLI 2.42.0
Windows PoSH

**Additional context**
Related to Azure Arc Jumpstart opened issue [ModuleNotFoundError: No module named 'azure.mgmt.web.v2021_01_01' #1549](https://github.com/microsoft/azure_arc/issues/1549)

/cc @Devinwong

**Comment by yonzhan** [comment]
route to CXP team

**Comment by navba-MSFT** [comment]
@likamrat Thanks for reaching out to us and reporting this issue. We are looking into this issue and we will provide an update.

**Comment by navba-MSFT** [comment]
@likamrat I was able to reproduce this issue on my local machine PS but not in Azure Cloud shell. While I am looking into this issue, Could you use Azure Cloud Shell as a workaround ?

**Comment by likamrat** [comment]
@navba-MSFT trying Cloud Shell as a workaround will not help. If you will look at the Jumpstart scenario, you will see our automation is running inside a client Windows Server VM in a PoSH session.

**Comment by navba-MSFT** [comment]
@likamrat Thanks for getting back. I have reached out to the product owners and discussing this issue with them. I will keep you posted about the progress.

**Comment by navba-MSFT** [comment]
@likamrat Does the problem persist if you uninstall/reinstall the extension (`az extension remove -n appservice-kube` then `az extension add -n appservice-kube`)?

**Comment by likamrat** [comment]
Yes. As part of our automation we always do a fresh install

**Comment by likamrat** [comment]
@navba-MSFT any update?

**Comment by jiasli** [comment]
This error is because API version `2021-01-01` is trimmed from `azure-mgmt-web` SDK by `scripts/trim_sdk.py` to reduce MSI package size (https://github.com/Azure/azure-cli/pull/23946):

https://github.com/Azure/azure-cli/blob/f952dc04ae2212d32bb8bb562fef40a7f749bacb/scripts/trim_sdk.py#L123

but `appservice-kube` extension hardcodes `2021-01-01`:

https://github.com/Azure/azure-cli-extensions/blob/f4033ee2830c68668f340e12c39d76b6d2b7ee3f/src/appservice-kube/azext_appservice_kube/custom.py#L300

```py
def _get_kube_client(cmd):
    client = web_client_factory(cmd.cli_ctx, api_version="2021-01-01")
    return client.kube_environments
```

https://github.com/Azure/azure-cli/pull/24017 removed hard coded app service API versions from main Azure CLI `appservice` command module, but the `appservice-kube` extension is left unchanged.

### Possible solutions

- `appservice-kube` extension can use the latest API version defined at https://github.com/Azure/azure-cli/blob/8c45b0feeaaa9a8ca3ed53c5cead1a8435b04fa6/src/azure-cli-core/azure/cli/core/profiles/_shared.py#L243
- If the above solution is not possible, `ResourceType.MGMT_APPSERVICE` can be defined as an `SDKProfile` so that different operations can use different API versions. This approach is adopted by `ResourceType.MGMT_COMPUTE`: https://github.com/Azure/azure-cli/blob/8c45b0feeaaa9a8ca3ed53c5cead1a8435b04fa6/src/azure-cli-core/azure/cli/core/profiles/_shared.py#L157-L159
- A less recommended solution is to define ad-hoc API version in `AD_HOC_API_VERSIONS`, but this is hard to maintain: https://github.com/Azure/azure-cli/blob/8c45b0feeaaa9a8ca3ed53c5cead1a8435b04fa6/src/azure-cli-core/azure/cli/core/profiles/_shared.py#L411-L415

**Comment by likamrat** [comment]
@jiasli @navba-MSFT any update on this?

**Comment by navba-MSFT** [comment]
@likamrat I hope you had a chance to look at the [above comment](https://github.com/Azure/azure-cli/issues/24744#issuecomment-1334688583) which has the root cause of this issue. The actual fix will take time to release ( no ETA at the moment). So the quick workaround is to uninstall your azure cli first and then re-install azure cli with `pip install azure-cli`. The mentioned issue only occurs in CLI msi package. So reinstalling the AzCLI should help.

**Comment by likamrat** [comment]
Will install older version help!

**Comment by navba-MSFT** [comment]
@likamrat Using most recent version is always _recommended_ to ensure that the CLI is up to date with all bug fixes. So please uninstall and reinstall the CLI as workaround.

**Comment by likamrat** [comment]
@navba-MSFT like I mentioned, in our automation we are always installing CLI from scratch so not sure why we keep coming back to uninstall/install.

Also, and again like I mentioned, pip install in a PowerShell environment is not trivial at all, especially for automated process.

Can we please get verification if an older version will work as a TEMPORARY workaround?

**Comment by navba-MSFT** [comment]
@likamrat Thanks for clarifying. I need to check this. I will get back to you once I have more details on this.

**Comment by likamrat** [comment]
@navba-MSFT I can confirm no issues with 2.40.0. I used the below.
```powershell
Invoke-WebRequest -Uri https://azurecliprod.blob.core.windows.net/msi/azure-cli-2.40.0.msi -OutFile .\AzureCLI.msi; Start-Process msiexec.exe -Wait -ArgumentList '/I AzureCLI.msi /quiet'; Remove-Item .\AzureCLI.msi
```

**Comment by likamrat** [comment]
Why is this marked as completed if we have no fix for latest CLI version?

**Comment by navba-MSFT** [comment]
@likamrat My Bad re-opening this issue again. As mentioned [above](https://github.com/Azure/azure-cli/issues/24744#issuecomment-1336034703), we are aware of this issue and its a work in progress with no ETA for the fix release. We will update this thread.

**Comment by navba-MSFT** [comment]
Adding the service team to update the thread once the fix is released.

@antcp, @AzureAppServiceCLI Could you please update this thread once you have the ETA for the fix ? Thanks in advance.

**Comment by likamrat** [comment]
@navba-MSFT @seligj95 any update on this?

**Comment by navba-MSFT** [comment]
@likamrat Since you are internal to Microsoft, I will loop you with the Product Owners in our internal ongoing email thread, if that helps.

**Comment by likamrat** [comment]
@navba-MSFT any updates on this?

**Comment by navba-MSFT** [comment]
@likamrat I had looped you in an email with the service team. Could you please follow-up on that and check for the status ?

**Comment by likamrat** [comment]
Looks like 2.45.0 no longer has the issue

**Comment by jaromirk** [comment]
I think I'm hitting the same issue on 2.47.0

```PowerShell
PS C:\Windows\system32> az version
{
  "azure-cli": "2.47.0",
  "azure-cli-core": "2.47.0",
  "azure-cli-telemetry": "1.0.8",
  "extensions": {
    "appservice-kube": "0.1.7",
    "connectedk8s": "1.3.16",
    "customlocation": "0.1.3",
    "k8s-configuration": "1.7.0",
    "k8s-extension": "1.4.0"
  }
}
PS C:\Windows\system32>
 
```

```PowerShell
PS C:\Windows\system32> az appservice kube show --resource-group $resourcegroup --name $kubeEnvironmentName | ConvertFrom-Json
WARNING: Command group 'appservice kube' is in preview and under development. Reference and support levels: https://aka.ms/CLI_refstatus
ERROR: The command failed with an unexpected error. Here is the traceback:
ERROR: No module named 'azure.mgmt.web.v2021_01_01'
Traceback (most recent call last):
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\knack/cli.py", line 233, in invoke
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 663, in execute
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 726, in _run_jobs_serially
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 697, in _run_job
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 333, in __call__
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/command_operation.py", line 363, in handler
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/arm.py", line 429, in show_exception_handler
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/command_operation.py", line 361, in handler
  File "C:\Users\LabAdmin\.azure\cliextensions\appservice-kube\azext_appservice_kube\custom.py", line 305, in show_kube_environments
    client = _get_kube_client(cmd)
  File "C:\Users\LabAdmin\.azure\cliextensions\appservice-kube\azext_appservice_kube\custom.py", line 301, in _get_kube_client
    return client.kube_environments
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/mgmt/web/_web_site_management_client.py", line 627, in kube_environments
ModuleNotFoundError: No module named 'azure.mgmt.web.v2021_01_01'
To check existing issues, please visit: https://github.com/Azure/azure-cli/issues
To open a new issue, please run `az feedback`
 
```

**Comment by rkacel** [comment]
Hi Im also facing this pb on Cloud shell as well 2.48.1

![image](https://user-images.githubusercontent.com/19334580/235324373-e600fd65-38dd-454e-a62e-b0399ac88000.png)
![image](https://user-images.githubusercontent.com/19334580/235324418-cbe57a10-d296-4611-8775-07a07ef772e7.png)

**Comment by tobyjrreu** [comment]
Same issue again with 2.48.1
`PS C:\_workingarea> az version
{
  "azure-cli": "2.48.1",
  "azure-cli-core": "2.48.1",
  "azure-cli-telemetry": "1.0.8",
  "extensions": {
    "appservice-kube": "0.1.7",
    "connectedk8s": "1.3.16",
    "customlocation": "0.1.3",
    "k8s-configuration": "1.7.0",
    "k8s-extension": "1.4.0"
  }
}
PS C:\_workingarea>`

**Comment by Juliehzl** [comment]
Please upgrade appservice-kube extension to 0.1.8, which should fix the issue.
